### PR TITLE
Fix missing imports due to pylance thinking open documents are under default workspace

### PR DESCRIPTION
**Pull request by bschnurr**
- inject workspace definitions into the "initialize" call to pylance so that there is no chance of a textdocument/didopen call triggering the creation of a dfeault workspace before getfolders is called to setup the workspace.

### CODE REVIEW

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs [hunk 1]
```diff
@@ -593,24 +590,35 @@ private Tuple<StreamData, bool> OnSendToServer(StreamData data) {
             return Tuple.Create(data, true);
         }
 
-        private async Task OnWorkspaceOpening(object sende, EventArgs e) {
+        private IEnumerable<WorkspaceFolder> GetFolders() {
```
Comment (rchiodo) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:593]: Could you have this always return a list? Every place you call it you're turning it into a list.

**Review by rchiodo — APPROVED** [review]

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs [hunk 2]
```diff
@@ -138,7 +135,7 @@ public async Task<Connection> ActivateAsync(CancellationToken token) {
             await JoinableTaskContext.Factory.SwitchToMainThreadAsync();
             CreateClientContexts();
 
-            _deferredSettingsChangedTimer = new Timer(OnDeferredSettingsChanged, state: null, Timeout.Infinite, Timeout.Infinite);
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:141]: just replacing this function with a lamda to cut down on fuctions

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs [hunk 3]
```diff
@@ -241,8 +238,16 @@ private async Task OnWorkspaceFolders(object sender, WorkspaceFolders.WorkspaceF
...
-            OnWorkspaceFolderWatched(this, EventArgs.Empty);
+            await TriggerWorkspaceUpdateConfig();
+        }
+
+        private Task TriggerWorkspaceUpdateConfig() {
+            Debug.WriteLine("Settings Changed");
+            return InvokeDidChangeConfigurationAsync(new LSP.DidChangeConfigurationParams() {
+                // Pylance will ask us for per workspace configuration settings. We can send
+                // default workspace settings here.
+                Settings = GetSettings()
```
Comment (bschnurr) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:249]: now include actual settings which get applied to any default workspace

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs (hunk 1 above)
Comment (bschnurr) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:593]: true. such a small list. no reason to make it enumerable

**Pull request merged**